tests/testsuite/rustflags.rs

```diff
@@ -993,6 +1054,188 @@ fn target_rustflags_normal_source() {
         .run();
 }
 
+#[cargo_test]
+fn target_rustflags_also_for_build_scripts() {
+    let p = project()
+        .file("src/lib.rs", "")
+        .file(
+            "build.rs",
+            r#"
+                fn main() { }
+                #[cfg(not(foo))]
+                fn main() { }
+            "#,
+        )
+        .file(
+            ".cargo/config",
+            &format!(
+                "
+            [target.{}]
+            rustflags = [\"--cfg=foo\"]
+            ",
+                rustc_host()
+            ),
+        )
+        .build();
+
+    p.cargo("build").run();
+}
```

Comment:
I believe that this is a regression relative to today, checking locally `target.$host.rustflags` is passed to build scripts on `cargo build`

Reply:
I think you're misreading the test — that's exactly what it's testing. Since `target.$host.rustflags` _does_ apply, `--cfg=foo` is passed to the build script, so the `cfg` gate for the second `main` is _not_ satisfied, so the build script compiles fine.

I can't say I blame you though — I've misread these so many times my self. I _hate_ that double-main trick as opposed to a good ol' fashioned `assert!(cfg!(foo));`. I'll update it and the other uses of this trick for `build.rs` in this file.

Reply:
Ah yes I think I misread the condition here, sorry about that.